collections/_article/whats-new-in-csharp-for-godot-4-0.md

```diff
@@ -0,0 +1,357 @@
+author: Raul Santos
+image: /storage/blog/covers/whats-new-in-csharp-for-godot-4-0.webp
+date: 2023-02-25 17:00:00
+---
+
+The Godot 4.0 release includes a lot of new features and improvements, check the release announcement for an overview on all the changes. In this article, we'll go into more detail about the C# specific changes.
+
+* [Runtime changes](#runtime-changes)
+* [Engine interop with source generators](#engine-interop-with-source-generators)
+* [The new Variant type](#the-new-variant-type)
+* [Collections changes](#collections-changes)
+* [Signals as events](#signals-as-events)
+* [Callable support](#callable-support)
+* [Int and Float changes](#int-and-float-changes)
+* [Renames](#renames)
+* [Exported property improvements](#exported-property-improvements)
+* [Improved C# documentation](#improved-c-documentation)
+* [NuGet packages](#nuget-packages)
+* [IDE support](#ide-support)
+* [Future](#future)
+  - [Global classes](#global-classes)
+  - [Mobile and web support](#mobile-and-web-support)
+  - [Full AOT and trimming support](#full-aot-and-trimming-support)
+  - [GDExtension support](#gdextension-support)
+  - [Improve Source Generators extensibility](#improve-source-generators-extensibility)
+  - [Editor unification](#editor-unification)
+  - [Godot as a library](#godot-as-a-library)
+
+<div class="card card-warning">
+  <p><strong>⚠️ Warning</strong></p>
+  <p>Projects created with a previous unstable version of Godot 4.0 will need you to delete the <code>res://.godot/mono</code> directory before opening them in the stable version of Godot 4.0, to ensure that no older assemblies are used.</p>
+  <p>Godot 3 projects can be converted to Godot 4.0 with the built-in project converter, but due to limitations of how the project converter works using regex, the converted project will likely require lots of manual fixing before it can be opened with Godot 4.0. We don't officially support opening Godot 3 C# projects with Godot 4.0 and recommend to create a new project from scratch.</p>
+</div>
+
+## Runtime changes
+
+The main change for C# support is that we moved away from the Mono SDK and we now use the **.NET SDK** to embed the .NET runtime, this means we use the CoreCLR runtime for desktop platforms, and in the future we would use the Mono runtime for mobile. Mobile and web support will be added in a future Godot release.
+
+With this change, the Mono module is now called **.NET module**, but the only .NET language officially supported is still **C#**. The Godot API is also now targeting .NET 6, the latest .NET LTS release.
+
+For end users, this means they just need to install the latest [.NET SDK](https://get.dot.net) to start working with C# in Godot. The default target for game projects will be .NET 6 allowing you to use the new modern C# features included in .NET that are not easily accessible in Godot 3. It should also be possible to target newer frameworks, although we can't guarantee it will work properly.
```

Comment:
"something that are" doesn't sound right to me since "something" is singular. The subject of "are not easily accessible" was meant to be the "modern C# features".